etl/pad/raw/dungeon.py

```diff
@@ -3,22 +3,80 @@
 """
 
 import csv
-import re
 from io import StringIO
-from typing import Any, List, Optional
+from typing import Any, Dict, List, Optional, Set, Union
 
 from pad.common import pad_util
 from pad.common.dungeon_types import RawDungeonType, RawRepeatDay
+from pad.common.pad_util import ghtime
 from pad.common.shared_types import DungeonId, SubDungeonId
 
 # The typical JSON file name for this data.
 FILE_NAME = 'download_dungeon_data.json'
 
 
+def maybe_int(num: str) -> Union[str, int]:
+    return int(num) if num.isnumeric() else num
+
+
+def default_int(modifiers: dict, key: str, default: int) -> int:
+    """This exists because there are so many god damn typos in GH data"""
+    try:
+        return int(modifiers.get(key, default))
+    except ValueError:
+        return default
+
+
+def parse_modifiers(mod_str: str) -> Dict[str, Union[bool, str]]:
+    mods = {}
+    for mod in mod_str.split('|'):
+        if ':' not in mod or mod.endswith(':'):
+            mods[mod.split(':')[0]] = True
+            continue
+        k, v = mod.split(':', 1)
+        mods[k] = v.strip()
+    return mods
+
+
+def merge_defaults(data, defaults):
+    return list(data) + defaults[len(data):]
+
+
+class FixedCard(pad_util.Printable):
+    """A fixed card on a subdungeon team."""
+
+    def __init__(self, data: str, order_idx):
+        data = [maybe_int(v) for v in data.rstrip(';').split(';')]
+        data = merge_defaults(data, [None, 99, 0, 0, 0, 99, 99])
+        (self.monster_id, self.level,
+         self.plus_hp, self.plus_atk, self.plus_rcv,
+         self.awakening_count, self.skill_level,
+         *rest) = data
+
+        # If the monster_id is 0, we have to make it None to show there's no monster here
```

Comment:
well mention the foreign key issue in the comment too bc we were going to use 0 except for that